gapir/cc/linux/gles_renderer.cpp

```diff
@@ -121,9 +122,17 @@ GlesRendererImpl::GlesRendererImpl(GlesRendererImpl* shared_context)
         , mSharedContext(shared_context != nullptr ? shared_context->mContext : 0)
         , mPbuffer(0) {
 
-    mDisplay = XOpenDisplay(nullptr);
-    if (mDisplay == nullptr) {
-        GAPID_FATAL("Unable to to open X display");
+    if (shared_context != nullptr) {
+        // Ensure that shared contexts also share X-display.
+        // Drivers are know to misbehave/crash without this.
+        mDisplay = shared_context->mDisplay;
+        mOwnsDisplay = false;
```

Comment:
You are right in terms of GL/EGL, but we always create extra "root" context just for the purpose of context sharing (which has never any GL commands issued to it).  This reason for it is simpler debugging, and to deal with the fact sometimes recreate contexts just to resize buffers (maybe we should stop doing that).